cluster-autoscaler/cloudprovider/azure/azure_agent_pool.go

```diff
@@ -209,6 +211,55 @@ func (as *AgentPool) TargetSize() (int, error) {
 	return int(size), nil
 }
 
+func (as *AgentPool) getAllSucceededAndFailedDeployments() (succeededAndFailedDeployments []resources.DeploymentExtended, err error) {
+	ctx, cancel := getContextWithCancel()
+	defer cancel()
+
+	succeededFilter := "provisioningState eq 'Succeeded' and provisioningState eq 'Failed'"
+	succeededAndFailedDeployments, err = as.manager.azClient.deploymentsClient.List(ctx, as.manager.config.ResourceGroup, succeededFilter, nil)
+	if err != nil {
+		return nil, nil
+	}
+
```

Comment:
could we move the filter here and only return the autoscaler managed deployments?